django/utils/http.py

```diff
@@ -293,12 +295,21 @@ def is_safe_url(url, host=None):
             url = force_text(url)
         except UnicodeDecodeError:
             return False
+    if allowed_hosts is None:
+        allowed_hosts = set()
+    if host:
+        warnings.warn(
```

Comment:
I added this at @charettes suggestion.  What alternative are you suggesting exactly?

* Go back to accepting both a string or an iterator? (In this case, are you suggesting we go back to the original argument name?)
* Keep both arguments in, just drop the deprecation warning?
* Something else?